storage/src/global_state/state/mod.rs

```diff
@@ -703,10 +703,18 @@ pub trait StateProvider {
                 };
                 let balance_holds = match request.balance_handling() {
                     BalanceHandling::Total => BTreeMap::new(),
-                    BalanceHandling::Available => match tc.get_balance_holds(purse_addr) {
-                        Ok(holds) => holds,
-                        Err(tce) => return tce.into(),
-                    },
+                    BalanceHandling::Available => {
+                        match tc.get_balance_hold_config(BalanceHoldAddrTag::Gas) {
+                            Ok(Some((block_time, _, interval))) => {
+                                match tc.get_balance_holds(purse_addr, block_time, interval) {
+                                    Ok(holds) => holds,
+                                    Err(tce) => return tce.into(),
+                                }
+                            }
+                            Ok(None) => BTreeMap::new(),
+                            Err(tce) => return tce.into(),
+                        }
+                    }
```

Comment:
Why are you changing this to only consider gas holds? I'm pretty sure it should consider all holds.